lib/plugin/mk-plugin.nix

```
@@ -0,0 +1,90 @@
+{
+  lib,
+  nixvimOptions,
+  nixvimUtils,
+}:
+with lib;
+config:
+{
+  name,
+  maintainers,
+  url ? defaultPackage.meta.homepage,
+  imports ? [ ],
+  description ? null,
+  # deprecations
+  optionsRenamedToSettings ? [ ],
+  # colorscheme
+  isColorscheme ? false,
+  colorscheme ? name,
+  # options
+  originalName ? name,
+  defaultPackage ? null,
+  extraOptions ? { },
+  # config
+  extraConfig ? cfg: { },
+  extraPlugins ? [ ],
+  extraPackages ? [ ],
+}:
+let
+  namespace = if isColorscheme then "colorschemes" else "plugins";
+
+  optionsRenamedToSettingsWarnings =
+    let
+      basePluginPath = [
+        namespace
+        name
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    map (
+      option:
+      let
+        optionPath = if isString option then [ option ] else option; # option is already a path (i.e. a list)
+
+        optionPathSnakeCase = map nixvimUtils.toSnakeCase optionPath;
+      in
+      mkRenamedOptionModule (
+        [
+          namespace
+          name
+        ]
+        ++ optionPath
+      ) (settingsPath ++ optionPathSnakeCase)
+    ) optionsRenamedToSettings;
+in
+{
+  meta = {
+    inherit maintainers;
+    nixvimInfo = {
+      inherit description url;
+      path = [
+        namespace
+        name
+      ];
```

```suggestion
      inherit description url path;
```

Let's define this in a wider scope or _as a default-arg_ so we can use it here **and** in the deprecations.